x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -163,11 +163,11 @@ export function addLayer(layerDescriptor) {
       return;
     }
 
-    dispatch({
+    await dispatch({
```

Comment:
This might've been an extra `await` that snuck in when I was doing some work to prevent timing issues when replacing the geojson temp layer with the newly indexed layer. I'm seeing no issues on removal. If any issues re-emerge, I'll look at some different approaches.